src/protocolgame.cpp

```diff
@@ -988,6 +988,26 @@ void ProtocolGame::parseRotateItem(NetworkMessage& msg)
 	addGameTaskTimed(DISPATCHER_TASK_EXPIRATION, &Game::playerRotateItem, player->getID(), pos, stackpos, spriteId);
 }
 
+void ProtocolGame::parseRuleViolationReport(NetworkMessage &msg)
+{
+	RuleViolationType_t report_type = (RuleViolationType_t) msg.getByte();
+	RuleViolationReasons_t report_reason = (RuleViolationReasons_t) msg.getByte();
+	const std::string target_name = msg.getString();
+	std::string player_comment = msg.getString();
+	std::string translation = "";
```

Comment:
@ranisalt you mean just std::string translation;? i'm not used to STL, does it already initialize as an empty string? an empty string is required here since report due unofficial software client doesn't send any translation, so i assumed it would be a lot of garbage values when i don't assign.